tests/components/my/test_init.py

```diff
@@ -0,0 +1,14 @@
+"""Test the my init."""
+
+from homeassistant.components.frontend import DATA_PANELS
+from homeassistant.components.my import URL_PATH
+from homeassistant.setup import async_setup_component
+
+
+async def test_setup(hass):
+    """Test setup."""
+    assert await async_setup_component(hass, "my", {"foo": "bar"})
+    panel = hass.data[DATA_PANELS][URL_PATH]
```

Comment:
Maybe patch `frontend.async_register_built_in_panel` instead and assert the mock call_args? We don't need to test the frontend integration code, just that we make the register call correctly.